packages/netlify-cms-core/src/actions/config.js

```diff
@@ -231,9 +250,17 @@ export function applyDefaults(config) {
 
           const files = collection.get('files');
           if (files) {
-            if (i18n && collection.has(I18N)) {
-              throw new Error('i18n configuration is not supported for files collection');
+            if (
```

Comment:
Can we extract this logic to a function, e.g. `isI18nAllowed`?
Also I think setting:
```yaml
file:
  i18n:
    structure: multiple_folders
```
Would pass this check